src/Microsoft.AspNetCore.SystemWebAdapters.CoreServices/SessionState/RemoteSession/RemoteAppSessionDispatcher.cs

```diff
@@ -0,0 +1,107 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.SystemWebAdapters.SessionState.RemoteSession;
+
+/// <summary>
+/// This is used to dispatch to either <see cref="DoubleConnectionRemoteAppSessionManager"/> if in read-only mode or <see cref="SingleConnectionWriteableRemoteAppSessionStateManager"/> if not.
+/// </summary>
+internal sealed partial class RemoteAppSessionDispatcher : ISessionManager
+{
+    private readonly IOptions<RemoteAppSessionStateClientOptions> _options;
+    private readonly ISessionManager _singleConnection;
+    private readonly ISessionManager _doubleConnection;
+    private readonly ILogger _logger;
+
+    public static ISessionManager Create(
```

Comment:
Wouldn't we always just get this from DI? Why do we need a static create method?

Reply:
For simplicity in testing, I want to be able to just pass in a mock so this facilitates that

Reply:
also added a comment to the method to make it clear what it's for